src/rgw/rgw_common.cc

```diff
@@ -1132,7 +1134,7 @@ bool verify_user_permission_no_policy(const DoutPrefixProvider* dpp, struct req_
                             const int perm)
 {
   /* S3 doesn't support account ACLs. */
-  if (!user_acl)
+  if ((s->user->type != TYPE_NONE || s->user->user_id == rgw_user(RGW_USER_ANON_ID)) && !user_acl)
```

Comment:
you can use `s->auth.identity->is_anonymous()` for the anon check

Reply:
Ok, thanks

Reply:
if the check is logically correct, please move any cleanups/syntax improvements to a new commit, so downstream is not affected

Reply:
The check is logically correct. I'll create a separate commit for what Casey has suggested.